### PR TITLE
Crop hints tutorial

### PR DESCRIPTION
**Pull request by gguuss**
Adds a Cloud client based tutorial to replace the existing Apiary one. Also changes snippets to include an aspect ratio so that the crop hints returned are not just the original image dimensions.

### CODE REVIEW

File: vision/cloud-client/detect.py [hunk 1]
```diff
@@ -356,7 +356,7 @@ def detect_crop_hints(path):
         content = image_file.read()
     image = vision_client.image(content=content)
 
-    hints = image.detect_crop_hints()
+    hints = image.detect_crop_hints({1.77})
```
Comment (jonparrott) [inline vision/cloud-client/detect.py]: Interesting that this method takes a set of length one?

File: vision/cloud-client/tutorials/crop_hints/crop_hints.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
+"""
```
Comment (jonparrott) [inline vision/cloud-client/tutorials/crop_hints/crop_hints.py]: Needs license header.

File: vision/cloud-client/tutorials/crop_hints/crop_hints.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+Crop Hints tutorial.
```
Comment (jonparrott) [inline vision/cloud-client/tutorials/crop_hints/crop_hints.py]: One-line summary goes on the same lines as ```""""```.

File: vision/cloud-client/tutorials/crop_hints/crop_hints.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+Run this script on an image to output a cropped image or an image highlighting
+suggested crop regions on an image:
+    python crop_hints.py <path-to-image> <crop | draw>
```
Comment (jonparrott) [inline vision/cloud-client/tutorials/crop_hints/crop_hints.py]: argparse will spit this out for you.

File: vision/cloud-client/tutorials/crop_hints/crop_hints.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+# [START imports]
+import argparse
+import io
+
+from google.cloud import vision
+from PIL import Image, ImageDraw
+# [END imports]
+
+
+def getCropHint(path):
```
Comment (jonparrott) [inline vision/cloud-client/tutorials/crop_hints/crop_hints.py]: `snake_case`.

File: vision/cloud-client/tutorials/crop_hints/crop_hints.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    # [START get_crop_hint]
+    """Detect crop hints on a single image and return the first result."""
+    vision_client = vision.Client()
+    with io.open(path, 'rb') as image_file:
```
Comment (jonparrott) [inline vision/cloud-client/tutorials/crop_hints/crop_hints.py]: blank newline above and below nested blocks, please.

File: vision/cloud-client/tutorials/crop_hints/crop_hints.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    draw.line([vects[1].x_coordinate, vects[1].y_coordinate,
+              vects[2].x_coordinate, vects[2].y_coordinate],
+              fill='red', width=3)
+    draw.line([vects[2].x_coordinate, vects[2].y_coordinate,
+              vects[3].x_coordinate, vects[3].y_coordinate],
+              fill='red', width=3)
+    draw.line([vects[3].x_coordinate, vects[3].y_coordinate,
+              vects[0].x_coordinate, vects[0].y_coordinate],
+              fill='red', width=3)
+    del draw
```
Comment (jonparrott) [inline vision/cloud-client/tutorials/crop_hints/crop_hints.py]: Is this necessary? It'll be garbage collected at the end of this function.

File: vision/cloud-client/tutorials/crop_hints/crop_hints.py [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    """Crop the image using the hints in the vector list."""
+    # [START crop_to_hint]
+    im = Image.open(image_file)
+    im2 = im.crop((vects[0].x_coordinate, vects[0].y_coordinate,
+                  vects[2].x_coordinate - 1, vects[2].y_coordinate - 1))
+    im2.save('output-crop.jpg', 'JPEG')
+    # [END crop_to_hint]
+
+
+def main(image_file, mode):
```
Comment (jonparrott) [inline vision/cloud-client/tutorials/crop_hints/crop_hints.py]: I'm not a huge fan of main functions, is it possible to restructure this to not need it?

File: vision/cloud-client/detect.py (hunk 1 above)
Comment (gguuss) [inline vision/cloud-client/detect.py]: I agree, this is specific to the hand-written veneers. The API takes an opaque settings object ([ImageContext](https://cloud.google.com/vision/docs/reference/rest/v1/images/annotate#ImageContext)) which is wrapped at the client library layer and is currently missing certain features.

File: vision/cloud-client/tutorials/crop_hints/crop_hints.py [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+    parser.add_argument('image_file', help='The image you\'d like to crop.')
+    parser.add_argument('mode', help='Set to "crop" or "draw".')
+    args = parser.parse_args()
+
+    parser = argparse.ArgumentParser()
+
+    if args.mode == 'crop':
+        crop_to_hint(args.image_file)
+
+    if args.mode == 'draw':
```
Comment (jonparrott) [inline vision/cloud-client/tutorials/crop_hints/crop_hints.py]: ```elif args.mode == 'draw':```, please.

**Comment by gguuss** [comment]
Addressed nit.

**Comment by jonparrott** [comment]
@dpebot will you merge when travis passes?

**Comment by jonparrott** [comment]
@gguuss looks like you need to add pillow to requirements.txt?

**Comment by gguuss** [comment]
pillow is there, `pillow==4.0.0` I had noticed the issue when I first pushed my branch -- maybe the CI is not installing it?

**Comment by jonparrott** [comment]
I can't see why that would happen, does it work locally?

**Pull request merged**